src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -804,4 +845,14 @@ public FontRenderContext getFontRenderContext() {
 		calcGfx.addRenderingHints(renderingHints);
 		return calcGfx.getFontRenderContext();
 	}
+
+	// In case a single 2D graphic object has to be written (one or more times)
+	// into a PDF document, this method isn't needed.
+	// If, instead, several 2D graphic objects have to be written (one or more times)
+	// into a PDF document, the graphic state has to be saved after a graphic object
+	// has been written for the last time and a new graphic object has to be created
+	public void saveGraphicsState() throws IOException
```

Comment:
I know that it deals with the internal content stream, but it is definitively not a NOP: if I try to print several different charts in the same PDF file, without that instruction I get a corrupted file; while it works if I save the graphic state after each chart print.
I guess PDFBox realeses some resources with saveGraphicState().
Maybe there's some different/better way to achieve the same result (?)